test/hybrid/server.dart

```diff
@@ -84,6 +84,8 @@ hybridMain(StreamChannel channel) async {
     request.headers.forEach((name, value) {
       // Ignore headers that are generated by the client
       if (_ignoreHeaders.contains(name)) return;
+      // Ignore a zero content-length header generated by the client
+      if (name == 'content-length' && value == '0') return;
```

Comment:
We definitely shouldn't be re-writing the headers on the server-side. That doesn't make any sense.

If underlying implementations consistently fill in a content length, though, I suppose it's fine for us to default to `0` rather than to not setting it.

Reply:
They don't consistently fill in a content length that's what makes this messy. You can see the test logs on this PR https://travis-ci.org/dart-lang/http/jobs/350628472 or run this branch locally yourself to see.

Reply:
I'm fine just always explicitly saying `content-length: 0` to provide a strong guarantee about when that will be set.

Reply:
Ok so roll back to https://github.com/dart-lang/http/pull/154/commits/036e4fd6fed9e6e9e222fa22541ed6df677e9146 then we're good to go?

Reply:
I think you mean 3269f49e94c5742d6e0a1ded3f7699c7b8193e85? If so, yeah.